common/src/main/java/com/wynntils/screens/maps/IconFilterScreen.java

```diff
@@ -0,0 +1,346 @@
+                            icons.replaceAll((key, value) -> true);
+                            button.active = false;
+                            excludeAllButton.active = true;
+                            populateIcons();
+                        })
+                .pos((int) backgroundX, (int) (dividedHeight * 3))
+                .size(filterButtonWidth, 20)
+                .build();
+
+        includeAllButton.active = icons.containsValue(false);
+
+        this.addRenderableWidget(includeAllButton);
+
+        excludeAllButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.iconFilter.excludeAll"), (button) -> {
+                            icons.replaceAll((key, value) -> false);
+                            button.active = false;
+                            includeAllButton.active = true;
+                            populateIcons();
+                        })
+                .pos((int) backgroundX + filterButtonWidth + 5, (int) (dividedHeight * 3))
+                .size(filterButtonWidth, 20)
+                .build();
+
+        excludeAllButton.active = icons.containsValue(true);
+
+        this.addRenderableWidget(excludeAllButton);
+        // endregion
+
+        // region done button
+        this.addRenderableWidget(
+                new Button.Builder(Component.translatable("screens.wynntils.iconFilter.done"), (button) -> McUtils.mc()
+                                .setScreen(previousScreen))
+                        .pos((int) (dividedWidth * 44), (int) (dividedHeight * 3))
+                        .size(filterButtonWidth, 20)
+                        .build());
+        // endregion
+
+        // The icons that will be displayed
+        usedIcons = new ArrayList<>(icons.keySet());
+
+        calculateIconSize();
+
+        populateIcons();
+    }
+
+    @Override
+    public void doRender(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        super.doRender(guiGraphics, mouseX, mouseY, partialTick);
+        renderScrollButton(guiGraphics.pose());
+
+        // Uncomment when editing UI
+        //                RenderUtils.renderDebugGrid(guiGraphics.pose(), GRID_DIVISIONS, dividedWidth, dividedHeight);
+    }
+
+    @Override
+    public void renderBackground(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        super.renderBackground(guiGraphics, mouseX, mouseY, partialTick);
+
+        RenderUtils.drawScalingTexturedRect(
+                guiGraphics.pose(),
+                Texture.WAYPOINT_MANAGER_BACKGROUND.resource(),
+                backgroundX,
+                backgroundY,
+                0,
+                backgroundWidth,
+                backgroundHeight,
+                Texture.WAYPOINT_MANAGER_BACKGROUND.width(),
+                Texture.WAYPOINT_MANAGER_BACKGROUND.height());
+    }
+
+    public void toggleIcon(Texture icon) {
```

Comment:
Let's keep the order of classes we usually use, that is:

- Constructors
- Overriden methods
- public
- protected
- private